ph5/clients/ph5tostationxml.py

```diff
@@ -222,6 +222,88 @@ def __init__(self, args):
+    
+        # Combine both by replace stage one in the data logger with stage
+        # one of the sensor.
+        dl_resp.response_stages.pop(0)
+        dl_resp.response_stages.insert(0, sensor_resp.response_stages[0])
+    
+        if not hasattr(dl_resp, "instrument_sensitivity"):
+            msg = "Could not find an instrument sensitivity - will not " \
+                "recalculate the overall sensitivity."
+            warnings.warn(msg)
+        elif not dl_resp.instrument_sensitivity.input_units:
+            msg = "Could not determine input units - will not " \
+                "recalculate the overall sensitivity."
+            warnings.warn(msg)
+        else:
+            i_u = dl_resp.instrument_sensitivity.input_units
+    
+            unit_map = {
+                "DISP": ["M"],
+                "VEL": ["M/S", "M/SEC"],
+                "ACC": ["M/S**2", "M/(S**2)", "M/SEC**2", "M/(SEC**2)",
+                        "M/S/S"]}
+            unit = None
+            for key, value in unit_map.items():
+                if i_u and i_u.upper() in value:
+                    unit = key
+            if not unit:
+                msg = ("Do not know how to map unit '%s' to "
+                       "displacement, velocity, or acceleration - overall "
+                       "sensitivity will not be recalculated.") % i_u
+                warnings.warn(msg)
+            else:
+                # lookup normalization frequency of sensor's poles-and-zeros
+                # stage, it should be in the flat part of the response
+                stage_one = dl_resp.response_stages[0]
+                try:
+                    frequency = stage_one.normalization_frequency
+                except AttributeError:
+                    frequency = None
+                for stage in dl_resp.response_stages[::-1]:
+                    # determine sampling rate
+                    try:
+                        sampling_rate = (
+                            stage.decimation_input_sample_rate /
+                            stage.decimation_factor)
+                        break
+                    except:
+                        continue
+                else:
+                    sampling_rate = None
+                if sampling_rate:
+                    # if sensor's normalization frequency is above 0.5*nyquist,
+                    # use that instead (e.g. to avoid computing an overall
+                    # sensitivity above nyquist)
+                    if frequency:
+                        nyquist = sampling_rate / 2.0
+                        frequency = min(frequency, nyquist / 2.0)
+                    else:
+                        frequency = nyquist / 2.0
+                freq, gain = \
+                    dl_resp._get_overall_sensitivity_and_gain(
+                        output=unit, frequency=frequency)
+                dl_resp.instrument_sensitivity.value = gain
+                dl_resp.instrument_sensitivity.frequency = freq

```

Comment:
Cool I'm setting up the PH5WS Beta Web Services to use this version of the code. If everything still works I think it is safe to merge.